### ISSUE TITLE
@:native  "." output "_" instead of "."

### ISSUE THREAD

**Issue by sonygod** [HaxeFoundation/haxe#4418]
@:native "." output "_"instead of "."

I really want to output "." instead "_"

like 

    @:native("BetterList$1.ctor")

      class BetterList
   {

	public function new(obj:Dynamic) 
	{
		
	}
	
   }

  Haxe will ouput  :  new BetterList$1_ctor  instead new BetterList$1.ctor

  how to ouput new BetterList$1.ctor?

**Comment by nadako** [comment]
It's probably cleaner to use `__init__` for that:

```haxe
class BetterList {
    public function new(obj:Dynamic)  {
    }

    static function __init__() {
        untyped __js__("BetterList$1.ctor = {0}", BetterList);
    }
}
```

Generates:
```js
var BetterList = function(obj) {
};
BetterList$1.ctor = BetterList;
```

**Comment by sonygod** [comment]
still not work
   @:extern class BetterList
   {

	public function new(obj:Dynamic) 
	{
		
	}
	 static function __init__() {
        untyped __js__("BetterList$1.ctor = {0}", BetterList);
      }


output  var bl = new BetterList(1);   instead new BetterList$1.ctor

**Comment by Simn** [comment]
I agree that `@:native` shouldn't go through any other renaming mechanisms.
